spec/statement.dd

```diff
@@ -984,30 +992,28 @@ $(CONSOLE
 8
 9
 )
-        $(P $(D ref) can not be applied to the index values.)
-
-        $(P If not specified, the $(I Type)s in the $(I ForeachType) can be
-        inferred from
-        the type of the $(I ForeachAggregate).
-        )
+        $(P $(D ref) cannot be applied to an array index variable.)
 
 $(H3 $(LNAME2 foreach_restrictions, Foreach Restrictions))
 
-        $(P The aggregate itself must not be resized, reallocated, free'd,
+        $(UNDEFINED_BEHAVIOR The aggregate itself must not be resized, reallocated, free'd,
```

Comment:
I'm not sure this is undefined behavior, that implies it must be prevented in `@safe` code by the compiler.

Reply:
OK, updated.

Reply:
I've extended the examples and moved this to #3205.